assembl/static/js/app/views/admin/adminMessageColumns.js

```diff
@@ -0,0 +1,303 @@
+  },
+  getIndex: function() {
+    return _.indexOf(this.model.collection.models, this.model);
+  },
+  serializeData: function() {
+    return {
+      column: this.model,
+      index: this.getIndex(),
+      collsize: this.model.collection.length,
+    };
+  },
+  onRender: function() {
+    this.showChildView(
+      "columnName",
+      new SimpleLangStringEditPanel({
+        model: this.model.get('name'),
+      }));
+  },
+  reorderColumnUp: function(ev) {
+    var index = this.getIndex(),
+        previousModel = this.model.collection.at(index-1);
+    this.model.set('previous_column', previousModel.get('previous_column'));
+    previousModel.set('previous_column', this.model.id);
+    this.model.collection.sort();
+    this.model.save(null, {success: function() {
+      previousModel.save();
+    }});
+
+    ev.preventDefault();
+  },
+  reorderColumnDown: function(ev) {
+    var index = this.getIndex(),
+        nextModel = this.model.collection.at(index+1);
+    nextModel.set('previous_column', this.model.get('previous_column'));
+    this.model.set('previous_column', nextModel.id);
+    this.model.collection.sort();
+    this.model.save();
+    nextModel.save();
+    ev.preventDefault();
+  },
+  deleteColumn: function(ev) {
+    var index = this.getIndex();
+    if (index < this.model.collection.length - 2) {
+      var nextModel = this.model.collection.at(index+1);
+      nextModel.set('previous_column', this.model.get('previous_column'));
+      nextModel.save();
+    }
+    this.model.destroy();
+    ev.preventDefault();
+  },
+  changeIdentifier: function(ev) {
+    this.model.set('message_classifier', ev.currentTarget.value);
+    this.model.save();
+    ev.preventDefault();
+  },
+  changeColor: function(ev) {
+    this.model.set('color', ev.currentTarget.value);
+    this.model.save();
+    ev.preventDefault();
+  },
+});
+
+
+/**
+ * The collections of columns to be seen on this idea
+ * @class app.views.adminMessageColumns.MessageColumnList
+ */
+var MessageColumnList = Marionette.CollectionView.extend({
+  constructor: function MessageColumnList() {
+    Marionette.CollectionView.apply(this, arguments);
+  },
+  initialize: function(options) {
+    this.options = options;
+  },
+  childView: MessageColumnView,
+});
+
+
+/**
+  * The admin modal view
+  * It is barely a simple container for the real view: AdminMessageColumnsPanel
+  * @class app.views.adminMessageColumns.AdminMessageColumnsModal
+  */
+var AdminMessageColumnsModal = Backbone.Modal.extend({
+  constructor: function AdminMessageColumnsModal() {
+    Backbone.Modal.apply(this, arguments);
+  },
+
+  template: '#tmpl-modalWithoutIframe',
+  className: 'modal-define-columns popin-wrapper',
+  cancelEl: '.close, .js_close',
+
+  ui: {
+    body: '.js_modal-body',
+  },
+
+  onRender: function() {
+    var resultView = new AdminMessageColumnsPanel({
+      model: this.model,
+    });
+    this.$(this.ui.body).html(resultView.render().el);
```

Comment:
@paolina92 , I did not see the change and re-did it. I hope I am not creating conflict.